packages/component-library/src/BarChart/BarChart.js

```diff
@@ -31,8 +32,13 @@ const BarChart = ({
   error
 }) => {
   const chartDomain = domain || getDefaultDomain(data, dataKey, dataValue);
+  const allDataPropertiesExist =
```

Comment:
Extract this logic to utils — 

`checkDataKeys` function that takes `( data, keysArray )` and returns an object `{ allKeysValid: bool, keyName1: { isValid, validKeys, invalidKeys } keyName2: { isValid, validKeys, invalidKeys } }`